Rubberduck.CodeAnalysis/QuickFixes/IsMissingOnInappropriateArgumentQuickFix.cs

```diff
@@ -0,0 +1,127 @@
+            rewriter.Replace(context, UninitializedComparisonForParameter(parameter));
+        }
+
+        private static readonly Dictionary<string, string> BaseTypeUninitializedValues = new Dictionary<string, string>
+        {
+            { Tokens.Boolean.ToUpper(), Tokens.False },
+            { Tokens.Byte.ToUpper(), "0" },
+            { Tokens.Currency.ToUpper(), "0" },
+            { Tokens.Date.ToUpper(), "CDate(0)" },
+            { Tokens.Decimal.ToUpper(), "0" },
+            { Tokens.Double.ToUpper(), "0" },
+            { Tokens.Integer.ToUpper(), "0" },
+            { Tokens.Long.ToUpper(), "0" },
+            { Tokens.LongLong.ToUpper(), "0" },
+            { Tokens.LongPtr.ToUpper(),  "0"  },
+            { Tokens.Single.ToUpper(), "0" },
+            { Tokens.String.ToUpper(), Tokens.vbNullString }
+        };
+
+        private string UninitializedComparisonForParameter(ParameterDeclaration parameter)
+        {
+            var type = parameter.AsTypeName?.ToUpper() ?? string.Empty;
+            if (string.IsNullOrEmpty(type))
+            {
+                type = parameter.HasTypeHint
+                    ? SymbolList.TypeHintToTypeName[parameter.TypeHint].ToUpper()
+                    : Tokens.Variant.ToUpper();
+            }
+
+            if (BaseTypeUninitializedValues.ContainsKey(type))
+            {
+                return $"{parameter.IdentifierName} = {BaseTypeUninitializedValues[type]}";
+            }
+
+            if (type.Equals(Tokens.Object, StringComparison.InvariantCultureIgnoreCase))
+            {
+                return $"{parameter.IdentifierName} Is {Tokens.Nothing}";
+            }
+
+            if (type.Equals(Tokens.Object, StringComparison.InvariantCultureIgnoreCase) || parameter.AsTypeDeclaration == null)
+            {
+                return $"IsEmpty({parameter.IdentifierName})";
+            }
+
+            switch (parameter.AsTypeDeclaration.DeclarationType)
+            {
+                case DeclarationType.ClassModule:
+                    return $"{parameter.IdentifierName} Is {Tokens.Nothing}";
+                case DeclarationType.Enumeration:
+                    var members = _state.DeclarationFinder.AllDeclarations.OfType<ValuedDeclaration>()
+                        .FirstOrDefault(decl =>
+                            ReferenceEquals(decl.ParentDeclaration, parameter.AsTypeDeclaration) &&
+                            decl.Expression.Equals("0"));
+                    return $"{parameter.IdentifierName} = {members?.IdentifierName ?? "0"}";
+                default:
+                    return $"IsError({parameter.IdentifierName})";
+            }
+        }
+
+        public override string Description(IInspectionResult result) => Resources.Inspections.QuickFixes.IsMissingOnInappropriateArgumentQuickFix;
+
+        public override bool CanFixInProcedure => true;
+        public override bool CanFixInModule => false;
+        public override bool CanFixInProject => false;
```

Comment:
would it be unsafe to allow fixing in module/project?

Reply:
I don't think it would be unsafe, but the QuickFix is fairly specific to how the function is being called and the signature of the procedure it's called from.  My thought was that it would be less likely to cause surprises if it was only applying to the currently selected procedure.